File: src/ledger.cpp
#include "ledger.h"

#include "itemprocessor.h"
#include "ledgeritem.h"

void Ledger::appendItem(std::shared_ptr<LedgerItem> item)
{
   m_items.push_back(item);
}

void Ledger::processItems(ItemProcessor& processor) const
{
   processor.start();
   for (auto it : m_items)
   {
      it->processItem(processor);
   }
   processor.finish();
}

File: src/main/java/cashpiles/ledger/Ledger.java
package cashpiles.ledger;

import java.util.ArrayList;

@SuppressWarnings("serial")
public class Ledger extends ArrayList<LedgerItem> {

	public void process(ItemProcessor processor) {
		for (var item : this) {
			item.process(processor);
		}
	}

}
